Further document publish command (#1914)
VirtusLab/scala-cli: website/docs/commands/publishing/publish.md

## Repository settings

A repository is required for the `publish` command, and might need other settings to work fine
(to pass credentials for example). See [Repositories](#repositories) for more information.

When publishing from you CI, we recommend letting `scala-cli publish setup`
setting those settings via using directives. When publishing from your local machine to Maven Central,
we recommend setting the repository via a `publish.repository` directive, and keeping your
Sonatype credentials in the Scala CLI settings, via commands such as

> ```bash

Reviewer: ````suggestion
```bash ignore
````

- the [Bouncy Castle library](https://www.bouncycastle.org) (default, recommended)
- the local `gpg` binary on your machine

A signing mechanism will be chosen based on options and directives specified,

> it can also be overriden with `--signer` with one of those values: `bc`, `gpg` or `none`.

Reviewer 2: Can you explain the meanings of the values `bc` and `gpg`? Also, what happens when the none option is selected?

Scala CLI can generate and keep a secret key for you. For that, create the key with

> ```sh

Reviewer: ````suggestion
```bash ignore
````

scala-cli --power config --create-gpg-key
```

This generated key kept in config will be used by default unless specified otherwise, e.g.:
- with directives:

>     ```bash

Reviewer: ````suggestion
    ```scala
````

    //> using publish.secretKey env:PGP_SECRET
    //> using publish.secretKeyPassword command:get_my_password
    ```

- with options:
    ```sh

>     scala-cli publish \

Reviewer: ````suggestion
    ```bash ignore
    scala-cli --power publish \
````

      --secret-key env:PGP_SECRET \
      --secret-key-password file:pgp_password.txt \
      …
    ```

> Since these values should be kept secret the options and directives accept the format documented [here](/docs/reference/password-options.md).

Reviewer: ```suggestion
Since these values should be kept secret, the options and directives accept the format documented [here](/docs/reference/password-options.md).
```

> Furthermore command line options accept an extra format - `config:…` for using config entries.

Reviewer 2: Since the default behavior is to retrieve values from the config, do we need to include the `config:..` syntax in this sentence?

> Furthermore command line options accept an extra format - `config:…` for using config entries.

Reviewer: ```suggestion
Furthermore, command line options accept an extra format: `config:…` for using config entries.
```

#### GPG

Using GPG to sign artifacts requires the `gpg` binary to be installed on your system.
A benefit of using `gpg` to sign artifacts over Bouncy Castle is: you can use keys from
your GPG key ring, or from external devices that GPG may support.

To enable signing with GPG, pass `--gpg-key *key_id*` on the command line

> or specify it by `//>using publish.gpgKey "key_id"`.

Reviewer: ```suggestion
or specify it with a `using` directive: `//>using publish.gpgKey "key_id"`.
```

### Maven Central

> The easiest way right now to publish to Maven Central Repository is to use

Reviewer: NIT
```suggestion
Right now the easiest way to publish to Maven Central Repository is to use
```

## Publishing

Once all the necessary settings are set, publish a Scala CLI project with a command
such as this one:

<ChainedSnippets>

> ```sh
> scala-cli publish .

Reviewer: ````suggestion
```bash ignore
scala-cli --power publish .
````
